src/Stateless/MethodDescription.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+
+namespace Stateless
+{
+    /// <summary>
+    /// Describes a method - either an action (activate, deactivate, etc.) or a transition guard
+    /// </summary>
+    public class MethodDescription
```

Comment:
Since this is only exposed in the reflection API, can we please move it to the _Stateless.Reflection_ namespace and go with the `*Info` naming pattern? I know it's used under the hood outside of that context, but for a consumer this is going to be part of the reflection subsystem only.

`InvocationInfo` was the best non-conflicting name I could come up with - perhaps we can go with that until a better option emerges?